conda/cli/main_config.py

```diff
@@ -309,6 +310,17 @@ def execute_config(args, parser):
 
     # config.rc_keys
     if not args.get:
+
+        # Add representers for enums.
+        # Because a representer cannot be added for the base Enum class (it must be added for
+        # each specific Enum subclass), and because of import rules), I don't know of a better
+        # location to do this.
+        def enum_representer(dumper, data):
+            return dumper.represent_str(str(data))
+
+        yaml.representer.RoundTripRepresenter.add_representer(SafetyChecks, enum_representer)
+        yaml.representer.RoundTripRepresenter.add_representer(PathConflict, enum_representer)
+
```

Comment:
Is it possible to move the PathConflict, SafetyChecks imports and then do the above in `conda/common/serialize.py` ?

Reply:
Import rules for `conda.common` are the most restrictive of all import rules.

https://github.com/conda/conda/blob/4.4.7/conda/common/__init__.py#L3-L9

Basically, *nothing* that's conda-specific should be in `common`.  `PathConflict` and `SafetyChecks` are both conda-specific domain objects.

The "right answer" here is that the representer should work for any `Enum`.  Unfortunately that appears to be a limitation of both `ruamel.yaml` and `pyyaml`.

Unfortunately, I think we're stuck with what's here.